File: src/main/java/xdcc/downloader/objects/Channel.java
/*
Copyright 2015-2017 Hermann Krumrey <hermann@krumreyh.com>

This file is part of jxdcc.

jxdcc is free software: you can redistribute it and/or modify
it under the terms of the GNU General Public License as published by
the Free Software Foundation, either version 3 of the License, or
(at your option) any later version.

jxdcc is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.

You should have received a copy of the GNU General Public License
along with jxdcc.  If not, see <http://www.gnu.org/licenses/>.
*/

package xdcc.downloader.objects;

public class Channel {

    private String name;
    private Bot[] botList;

    public Channel(Server server, String name) {
        this.name = name;
        this.botList = new Bot[0];
    }

    public void addBot(Bot bot) {
        Bot[] newBotList = new Bot[this.botList.length + 1];
        for (int i = 0; i < this.botList.length; i++) {
            newBotList[i] = this.botList[i];
        }
        newBotList[this.botList.length] = bot;
        this.botList = newBotList;
    }

    public Bot[] getBotList() {
        return this.botList;
    }

    public String getName() {
        return this.name;
    }
}

File: hexchat-downloader/c/Objects/Channel.cpp
/*
Copyright 2015-2017 Hermann Krumrey <hermann@krumreyh.com>

This file is part of xdcc-downloader.

xdcc-downloader is free software: you can redistribute it and/or modify
it under the terms of the GNU General Public License as published by
the Free Software Foundation, either version 3 of the License, or
(at your option) any later version.

xdcc-downloader is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.

You should have received a copy of the GNU General Public License
along with xdcc-downloader.  If not, see <http://www.gnu.org/licenses/>.
*/

/*
 * @author Hermann Krumrey <hermann@krumreyh.com>
 */
#include "ServerList.h"

//public
//Constructor

/**
 * Creates a new Channel Object with a given name
 * @param name - the name of the channel
 */
Channel::Channel(string name) {

    this->name = name;

}

/**
 * Adds a bot to the channel
 * @param bot - the bot to be added
 */
void Channel::addBot(Bot bot) {

    this->bots.push_back(bot);

}

/**
 * Adds a pack to a bot belonging to the channel
 * @param pack - the pack to be added
 * @param botIndex - the index of the bot to which this pack should be added
 */
void Channel::addPack(Pack pack, int botIndex) {

    this->bots[botIndex].addPack(pack);

}

//Getter/Setter

string Channel::getName() {
    return this->name;
}

vector<Bot> Channel::getBots() {
    return this->bots;
}